sssd_test_framework/utils/ipa_ca.py

```diff
@@ -0,0 +1,264 @@
+
+        :param key_path: Path to save the private key.
+        :type key_path: str
+        :param csr_path: Path to save the CSR file.
+        :type csr_path: str
+        :param subject: Subject for the CSR (e.g., /CN=example).
+        :type subject: str
+        :param key_size: RSA key size in bits.
+        :type key_size: int
+        :raises RuntimeError: If CSR generation fails.
+        """
+        args: CLIBuilderArgs = {
+            "newkey": (self.cli.option.VALUE, f"rsa:{key_size}"),
+            "nodes": (self.cli.option.SWITCH, True),
+            "keyout": (self.cli.option.VALUE, key_path),
+            "out": (self.cli.option.VALUE, csr_path),
+            "subj": (self.cli.option.VALUE, subject),
+        }
+        result = self.host.conn.run(self.cli.command("openssl req", args), raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"OpenSSL CSR generation failed: {result.stderr}")
+
+    def _get_cert_serial(self, cert_path: str) -> str:
+        """
+        Extract the certificate serial number using OpenSSL.
+
+        :param cert_path: Path to the certificate file.
+        :type cert_path: str
+        :returns: The certificate serial number as a lowercase hex string.
+        :rtype: str
+        :raises RuntimeError: If serial extraction fails.
+        """
+        cmd = ["openssl", "x509", "-in", cert_path, "-noout", "-serial"]
+        cmdline = " ".join(shlex.quote(p) for p in cmd)
+        result = self.host.conn.run(cmdline, raise_on_error=False)
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to get serial from certificate: {result.stderr}")
+        out = (result.stdout or "").strip()
+        if "=" in out:
+            return out.split("=", 1)[1].lower()
+        return out.lower()
+
+    def _revocation_reason_to_code(self, reason: RevocationReason) -> int:
+        """
+        Map a revocation reason string to its corresponding numeric code.
+
+        :param reason: Revocation reason string.
+        :type reason: RevocationReason
+        :returns: Numeric reason code.
+        :rtype: int
+        """
+        reason_map = {
+            "unspecified": 0,
+            "key_compromise": 1,
+            "ca_compromise": 2,
+            "affiliation_changed": 3,
+            "superseded": 4,
+            "cessation_of_operation": 5,
+            "certificate_hold": 6,
+            "remove_from_crl": 8,
+            "privilege_withdrawn": 9,
+            "aa_compromise": 10,
+        }
+        return reason_map[reason]
+
+    def _parse_cert_info(self, output: str) -> Dict[str, str]:
+        """
+        Parse ipa cert-show output into a dictionary.
+
+        :param output: Raw command output from ipa cert-show.
+        :type output: str
+        :returns: Dictionary of certificate attributes.
+        :rtype: dict[str, str]
+        """
+        info: Dict[str, str] = {}
+        for line in (output or "").splitlines():
```

Comment:
See if you can re-use parse_attr from misc/__init__.py it does take output that is formatted "atttribute: value" and returns a dictionary.